tests/test_filesystems.py

```diff
@@ -87,3 +87,12 @@ async def test_resolve_path(self):
         assert fs._resolve_path(base) == base + "/"
         assert fs._resolve_path(f"{base}/subdir") == f"{base}/subdir"
         assert fs._resolve_path("subdirectory") == f"{base}/subdirectory"
+
+
+class TestGitHub:
+    async def test_subprocess_errors_are_surfaced(self):
```

Comment:
`get_directory` has quite a bit of branching, might be nice to cover some of those other paths.

Reply:
@bunchesofdonald makes sense, I added a few critical path tests